lib/diaspora/federation/entities.rb

```diff
@@ -200,37 +177,29 @@ def self.reshare(reshare)
         )
       end
 
-      def self.retraction(target)
+      def self.retraction(retraction)
+        case retraction.data[:target_type]
+        when "Contact"
```

Comment:
Hmm, so maybe we need a `RetractionDispatcher` or so?

But I still think that's out of scope here, this PR is for the protocol change and 0.2.0 version of the gem. The `ContactRetraction` refactoring was about something changed in this PR, so I added it here. But for the "retraction dispatching" refactoring I'd rather do/discuss it independent of this PR.